Main.hs

```diff
@@ -153,6 +160,19 @@ updatePackageSpec = execStateT $ do
       _ -> pure ()
       )
 
+    -- If the fetcher attribute is not set, we try to infer its value based on the url suffix
+    (,) <$> getPackageSpecAttr "fetcher" <*> getPackageSpecAttr "url" >>= \case
+      -- If a fetcher is set, we'll use it
+      (Just _, _) -> pure ()
+      -- We need an url to infer a fetcher
+      (_, Nothing) -> pure ()
+      (Nothing, Just (Aeson.String url)) -> do
+         let fetcher = if "tar.gz" `T.isSuffixOf` url
+                       then "fetchTarball"
+                       else "fetchurl"
+         setPackageSpecAttr "fetcher" (Aeson.String $ T.pack fetcher)
+      (_, _) -> pure ()
```

Comment:
isn't this branch redundant?

Reply:
For ghc, it is not:) I added a comment to explain we ignore the `fetcher` if the JSON value is not a string.